### DISCUSSION TITLE
How to mark a workflow as successful even if a step of it runs overtime?

### DISCUSSION THREAD

**Discussion by zaikunzhang** [community/community#58716] [Actions] [answered]
### Select Topic Area

Question

### Body

Dear all,

I have a workflow containing a step that is expected to take a long time to run. I am OK if this step runs overtime, as long as no error is raised during the step. However, my workflow is always marked as a failure when it takes more than six hours. 

Question: How to mark the workflow as successful even if (a step of) it runs overtime?

A concrete example: https://github.com/equipez/prima/blob/main/.github/workflows/stress_test.yml

This workflow is a **stress test** of a numerical package on excessively large problems. The purpose is to verify that it does not crash (due to memory violation, etc) before it is terminated manually even on such large problems. If the test runs for several hours without crashing, I will regard it as a success. 

As you can see, the long-running step in my workflow is the last step, which runs MATLAB code using [matlab-actions/run-command](https://github.com/matlab-actions/run-command) .  Therefore, I cannot use the bash `timeout` command to control the running time. 

Thank you very much for any comments or insights. 

Best regards,
Zaikun

**Comment by datpmt** [comment]
![image](https://github.com/community/community/assets/76593070/f40308e7-f436-43ea-90bf-1646197c9c31)
try this
```
jobs:
  test:
    name: Stress test of PRIMA on large problem.
    runs-on: ${{ matrix.os }}
    timeout-minutes: 720
    strategy:
      fail-fast: false
      matrix:
        os: [ubuntu-latest]
        matlab: [latest]
        solver: [cobyla, uobyqa, newuoa, bobyqa, lincoa]
        test: [normal, tough]

    steps:
      - name: Check out repository
        uses: actions/checkout@v3.5.3
        with:
          ssh-key: ${{ secrets.SSH_PRIVATE_KEY_ACT }} # This forces checkout to use SSH, not HTTPS
          submodules: recursive

      - name: Link gfortran for MATLAB on Linux
        if: startsWith(matrix.os, 'ubuntu')
        run: bash .github/scripts/link_gfortran ${{ env.GFORTRAN_VERSION }}

      - name: Cache MATLAB  # N.B.: Clear the cache when the `latest` version of MATLAB changes in March and September
        uses: actions/cache@v3.3.1
        with:
          path: ${{ runner.tool_cache }}/MATLAB
          key: ${{ matrix.os }}-${{ matrix.matlab }}-yes

      - name: Set up MATLAB with optimization toolbox
        uses: matlab-actions/setup-matlab@v2-beta
        with:
          release: ${{ matrix.matlab }}
          products: Optimization_Toolbox

      - name: Conduct the test
        uses: matlab-actions/run-command@v1.2.0
        with:
          command: |
            ver;
            root_dir = pwd();
            cd(fullfile(root_dir, 'matlab/tests'));
            options = struct();
            options.tough = strcmp('${{ matrix.test }}', 'tough');
            stress('${{ matrix.solver }}', options);
```

You can try setting: `timeout-minutes: 720`

**Reply by zaikunzhang** [reply]
Hello @datpmt ,

Thank you for your suggestion. However,

1. Even if  `timeout-minutes` is set to an extremely large value, my test will run over time. 
2. According to my communication with GitHub support, it is impossible to set `timeout-minutes` to a value higher than 360 for GitHub hosted runners. 
> Hi Zaikun,

> >So, for a workflow, timeout-minutes: x will take effect only if x < 360, right?

> Yes, that's the case for GitHub-hosted runners.
> Regards,
>GitHub Support

Thanks,
Zaikun

**Comment by zaikunzhang** [comment, marked as answer]
See here for a solution: 

https://github.com/libprima/prima/blob/main/.github/workflows/stress_test_matlab.yml#L175-L190

where the real business is done by the following script: 

https://github.com/equipez/github_actions_scripts/blob/main/check_success_timeout

Disclaimer: This solution is partially provided by GitHub Support.

**Reply by zaikunzhang** [reply]
N.B.: Remember to specify **`continue-on-error: true`** for the jobs of the long-running tests.
